Update interpolated-strings.md [RFC FS-1132] (#35195)
dotnet/docs: docs/fsharp/language-reference/interpolated-strings.md

Note that the type annotation must be on the interpolated string expression itself. F# does not implicitly convert an interpolated string into a <xref:System.FormattableString>.

## Extended syntax for string interpolation

When working with text containing multiple `{`, `}` or `%` characters already, extended string interpolation syntax can be used to remove the need for escaping.

Triple quote string literals can start with multiple `$` characters, which changes how many braces are required to open and close interpolation.
In these string literals, `{` and `}` characters do not need to be escaped:

```fsharp
let str = $$"""A string containing some {curly braces} and an {{"F#" + " " + "expression"}}."""
// "A string containing some {curly braces} and an F# expression."
let another = $$$"""A string with pairs of {{ and }} characters and {{{ "an F# expression" }}}."""
// "A string with pairs of {{ and }} characters and an F# expression."""
```

Number of `%` characters needed for format specifiers is affected in the same way:

> ```fsharp

Author: ```suggestion

```fsharp
```

Note that the type annotation must be on the interpolated string expression itself. F# does not implicitly convert an interpolated string into a <xref:System.FormattableString>.

## Extended syntax for string interpolation

> When working with text containing multiple `{`, `}` or `%` characters already, extended string interpolation syntax can be used to remove the need for escaping.

Reviewer: ```suggestion
When you work with text containing multiple `{`, `}` or `%` characters already, you can use extended string interpolation syntax to remove the need for escaping.
```

Triple quote string literals can start with multiple `$` characters, which changes how many braces are required to open and close interpolation.

> In these string literals, `{` and `}` characters do not need to be escaped:

Reviewer: ```suggestion
In these string literals, `{` and `}` characters don't need to be escaped:
```

```fsharp
let str = $$"""A string containing some {curly braces} and an {{"F#" + " " + "expression"}}."""
// "A string containing some {curly braces} and an F# expression."
let another = $$$"""A string with pairs of {{ and }} characters and {{{ "an F# expression" }}}."""
// "A string with pairs of {{ and }} characters and an F# expression."""
```

> Number of `%` characters needed for format specifiers is affected in the same way:

Reviewer: ```suggestion
The number of `%` characters needed for format specifiers is affected in the same way:
```
